### ISSUE TITLE
empty: add integers conversion

### ISSUE THREAD

**Issue by ccoVeille** [Antonboom/testifylint#115]
This pattern should be detected

```go
assert.Equal(t, 0, int(ret.ExitCode))
assert.Equal(t, 0, int(ret.GasUsed))
```
Example: https://github.com/ipfs-force-community/go-fvm-sdk/blob/d49236897af1fdb5e0f7d29573fcbe1d29e2214d/sdk/cases/send.go#L40-L41

The following code should be suggested
```go
assert.Empty(t, ret.ExitCode)
assert.Empty(t, ret.GasUsed)
```

This pattern is currently not detected, while it's pretty commonly used
https://github.com/search?q=language%3Ago+%22assert.Equal%28t%2C+0%2C+int%22&type=code

**Comment by Antonboom** [comment]
Please, do not use `Empty` for objects for which `len` is not applicable.
For me these are `Zero` cases and use `Empty` is not best and doubtful option here.

**Comment by ccoVeille** [comment]
That's right. As you declined the idea of a zero checker, I assumed you didn't want to suggest using zero.

BTW, I just found out these cannot be part of `empty` checker, as nothing detects this now

```go
a := 17
assert.Equal(t, 0, a)
```

So I'm unsure where to add it

**Comment by ccoVeille** [comment]
here are some other use cases

https://github.com/search?q=%22assert.Equal%28t%2C+int%280%29%2C%22+NOT+is%3Aarchived+NOT+is%3Afork+language%3Ago&type=code

https://github.com/karim7262/datadog-agent/blob/a48b9b58fb9f13e53782a936783da2cc4b7b0c7c/pkg/autodiscovery/providers/zookeeper_test.go#L206
```go
assert.Equal(t, int(0), zkr.cache.NumAdTemplates)
```

**Comment by Antonboom** [comment]
> As you declined the idea of a zero checker, I assumed you didn't want to suggest using zero.

Yes, 


https://github.com/Antonboom/testifylint/issues/75#issuecomment-2067582114
